github/Auth.py

```diff
@@ -307,7 +307,8 @@ def _is_expired(self) -> bool:
             self.__installation_authorization.expires_at
             - TOKEN_REFRESH_THRESHOLD_TIMEDELTA
         )
-        return token_expires_at < datetime.now(timezone.utc)
+        # to be fixed by https://github.com/PyGithub/PyGithub/pull/1831
+        return token_expires_at < datetime.now(timezone.utc).replace(tzinfo=None)
```

Comment:
Sorry, I'm confused. Why are you creating a date with the UTC timezone, then removing the timezone information?

Reply:
We need the UTC timestamp here (without timezone information), not the local time (without timezone information). Otherwise under- or overshoot the expiry by the local timezone offset hours.

The `token_expires_at` is a UTC timestamp without timezone information, so we cannot compare it with a timestamp that has timezone information.

Reply:
I understand now, thanks!